### ISSUE TITLE
Environment Variables don't seem to work.

### ISSUE THREAD

**Issue by Melechtna** [AsherGlick/Burrito#57]
I have a rather complicated setup to allow me to run GW2 through steam using lutris so that I can use a controller to play the game. I have all the variables stored in a bash script, which is kind of required to use it all correctly, and my current work around is to simply duplicate this script and run it manually in order to get the game link working. I have the autolaunch plugged with the same environment variables, but it's not launching on its own.

I'm running the latest git.

export` SDL_VIDEO_FULLSCREEN_DISPLAY="off"
export STEAM_RUNTIME="/home/melechtna/.local/share/lutris/runtime/steam"
export LD_LIBRARY_PATH="/home/melechtna/.local/share/lutris/runners/wine/lutris-GE-Proton7-7-x86_64/lib:/home/melechtna/.local/share/lutris/runners/wine/lutris-GE-Proton7-7-x86_64/lib64:/usr/lib/libfakeroot:/usr/lib/libstrangle/lib32:/usr/lib/libstrangle/lib64:/usr/lib/openmpi:/usr/lib32:/usr/lib:/usr/lib64:/home/melechtna/.local/share/lutris/runtime/lib32:/home/melechtna/.local/share/lutris/runtime/steam/i386/lib/i386-linux-gnu:/home/melechtna/.local/share/lutris/runtime/steam/i386/lib:/home/melechtna/.local/share/lutris/runtime/steam/i386/usr/lib/i386-linux-gnu:/home/melechtna/.local/share/lutris/runtime/steam/i386/usr/lib:/home/melechtna/.local/share/lutris/runtime/lib64:/home/melechtna/.local/share/lutris/runtime/steam/amd64/lib/x86_64-linux-gnu:/home/melechtna/.local/share/lutris/runtime/steam/amd64/lib:/home/melechtna/.local/share/lutris/runtime/steam/amd64/usr/lib/x86_64-linux-gnu:/home/melechtna/.local/share/lutris/runtime/steam/amd64/usr/lib:$LD_LIBRARY_PATH"
export DXVK_HUD="0"
export DXVK_LOG_LEVEL="none"
export STAGING_SHARED_MEMORY="1"
export __GL_SHADER_DISK_CACHE_PATH="/home/melechtna/Games/guild-wars-2"
export WINEDEBUG="-all"
export WINEARCH="win64"
export WINE="/home/melechtna/.local/share/lutris/runners/wine/lutris-GE-Proton7-7-x86_64/bin/wine"
export GST_PLUGIN_SYSTEM_PATH_1_0="/home/melechtna/.local/share/lutris/runners/wine/lutris-GE-Proton7-7-x86_64/lib64/gstreamer-1.0/:/home/melechtna/.local/share/lutris/runners/wine/lutris-fshack-6.14-3-x86_64/lib/gstreamer-1.0/"
export WINEPREFIX="/home/melechtna/Games/guild-wars-2"
export WINEESYNC="1"
export WINEFSYNC="1"
export WINEDLLOVERRIDES="d3d10core,d3d11,d3d12,d3d9,dxgi=n;winemenubuilder.exe=d"
export WINE_LARGE_ADDRESS_AWARE="1"
export game_name="Guild Wars 2"
export PYTHONPATH="/usr/lib/lutris:/usr/bin:/usr/lib/python39.zip:/usr/lib/python3.9:/usr/lib/python3.9/lib-dynload:/home/melechtna/.local/lib/python3.9/site-packages:/usr/lib/python3.9/site-packages"

LUTRIS_WINE="/home/melechtna/.local/share/lutris/runners/wine/lutris-GE-Proton7-7-x86_64/bin/wine"
GAME_EXE="/home/melechtna/src/Burrito/output/burrito_link.exe"

"$LUTRIS_WINE" "$GAME_EXE"
exit 1

**Comment by AsherGlick** [comment]
What do you have saved exactly in the `Wine Path` and `Environment Vars` sections of the settings. `Environment Vars` is not a shell script and each of those vars are parsed internally and then set, I should think about adding some sort of warning if parsing does not work. I would have to double check this but I also believe that `Wine Path` will not respect env vars so you would have to set that to be the value of `$LUTRIS_WINE` instead of a reference to the variable.

**Comment by Melechtna** [comment]
> What do you have saved exactly in the `Wine Path` and `Environment Vars` sections of the settings. `Environment Vars` is not a shell script and each of those vars are parsed internally and then set, I should think about adding some sort of warning if parsing does not work. I would have to double check this but I also believe that `Wine Path` will not respect env vars so you would have to set that to be the value of `$LUTRIS_WINE` instead of a reference to the variable.

That's the entire script, I just have the exports pasted in the environment variables.

**Comment by AsherGlick** [comment]
And you have `/home/melechtna/.local/share/lutris/runners/wine/lutris-GE-Proton7-7-x86_64/bin/wine` in the `Wine Path` ?

**Comment by Melechtna** [comment]
> And you have `/home/melechtna/.local/share/lutris/runners/wine/lutris-GE-Proton7-7-x86_64/bin/wine` in the `Wine Path` ?

Correct

**Comment by Melechtna** [comment]
![Screenshot_20220401_182103](https://user-images.githubusercontent.com/81296467/161353636-4b19d3f2-5d69-4b4b-b50a-bcf5580586b1.png)
Here's a screenshot of how I have it currently, I removed the word "export" from the beginning earlier to see if that would change anything, it didn't.

**Comment by AsherGlick** [comment]
Ok my last guess here is that `burrito_link.exe` is missing or in the wrong place somehow. Is it at

 `./burrito_link/burrito_link.exe`

relative to the cwd of the burrito binary or project (depending on how you run burrito)?

**Comment by Melechtna** [comment]
> Ok my last guess here is that `burrito_link.exe` is missing or in the wrong place somehow. Is it at
> 
> `./burrito_link/burrito_link.exe`
> 
> relative to the cwd of the burrito binary or project (depending on how you run burrito)?

It's in the same directory as burrito.x86_64, should it not be? Everything is in the same folder, with a Data folder inside said folder.

**Comment by AsherGlick** [comment]
I am guessing I placed it in a subfolder so that it could be accessed via the same relative path from the godot project directory, and the built executable directory. Unlike the GDNative libraries, `burrito_link.exe` is not automatically tracked. The CI packaging places it in the correct folder so this should not be an issue for users who are not building from source. I am going to chalk this up to a duplicate of #42 at this point.

Glad we got it resolved.
